### PR TITLE
fix: ⚡️ Fix dependency vulnerabilities

### LINKED ISSUE #26: [bug] "random" button doesn't respond
the "random" button on the teia main feed prints out a blank feed.
(since the UI update)

### PR DESCRIPTION
**Pull request by melMass**
## Disclaimer

This is updating a lot of dependencies. To properly test it locally if you have already pulled the repo before, you must first delete `node_modules` and re-run `npm i`

---
Current `main`:
```ruby
50 vulnerabilities (27 moderate, 20 high, 3 critical)
```

This PR:
```ruby
10 vulnerabilities (7 moderate, 3 high)
```

---
I'm following dependabot alerts for now.
Most of them actually came from the templates.

Fixes:
[dependabot/2](https://github.com/teia-community/teia-ui/security/dependabot/2)
[dependabot/9](https://github.com/teia-community/teia-ui/security/dependabot/9)
[dependabot/50](https://github.com/teia-community/teia-ui/security/dependabot/50)
[dependabot/26](https://github.com/teia-community/teia-ui/security/dependabot/26)

### CODE REVIEW

File: .env
Comment (melMass) [inline .env:1]: This is due to that: 
https://github.com/facebook/create-react-app/discussions/11767

File: src/utils/verify.js
Comment (melMass) [inline src/utils/verify.js]: I extracted the functions we make us of only from the ez lib thing which had many security issues associated. 

This seems to still error out. I need to investigate further

**Comment by melMass** [comment]
(it's not ready for review, I'm trying to trigger the action)

**Comment by melMass** [comment]
Ready to review.
IMO we should merge it and not Squash it because there is a lot going on, I tried to separate the commits properly

File: src/utils/verify.js
Comment (melMass) [reply src/utils/verify.js]: Solved in 2ec678975c6f9ca07d77dc4ef4e72db6b8725462

**Comment by melMass** [comment]
@xat Do you know why there is no action running anymore?

**Comment by xat** [comment]
hmm, the action did work in the pr that i just opened. maybe a temp problem?

**Comment by xat** [comment]
@melMass i would suggest that we wait a week or so, before merging this one after we ironed out the more pressing bugs that got revealed due to the bigger release. because updating dependencies can bring in further instability, which i would want to avoid until things are bit more settled.

would it make sense to move the "get random objkts" commit into a separate pr, then we could merge that one upfront?

what do you think?

**Comment by melMass** [comment]
> @melMass i would suggest that we wait a week or so, before merging this one after we ironed out the more pressing bugs that got revealed due to the bigger release.


Yes it makes sense, I spent hours on it so I won't have the same time to dedicate fixing it in a week, but we could use it as a reference or something like that 👌

**Comment by xat** [comment]
@melMass ah ok, so delaying the merge would create extra effort.

hmm, do you think it is safe enough to merge it sooner? my feeling is that it does require some extended testing.

**Comment by melMass** [comment]
> @melMass ah ok, so delaying the merge would create extra effort.
> 
> 
> 
> hmm, do you think it is safe enough to merge it sooner? my feeling is that it does require some extended testing.

Yes it definitely need more extensive testing, but I meant more if we fix other stuffs under it it might be better to start this one over, following it as a ref. I spent most of my time checking breaking changes in libraries, I know them now

**Comment by melMass** [comment]
Rebased 😏  🤣

I'm going to try to maintain it as a virtual `develop` branch and rebase regularly until someone can look into it/ we can test it further

File: craco.config.js
Comment (xat) [inline craco.config.js:5]: nice 👍

File: package.json
Comment (xat) [inline package.json:9]: did you check if there were any breaking changes?

File: package.json
Comment (xat) [inline package.json]: is this used?

File: package.json
Comment (xat) [inline package.json:68]: why were these libraries not needed before?

File: src/components/collab/create/CollaboratorRow.js
Comment (xat) [inline src/components/collab/create/CollaboratorRow.js:3]: import can be removed

File: src/data/hicdex.js
Comment (xat) [inline src/data/hicdex.js]: console.logs in the file probably can be removed

File: src/data/hicdex.js
Comment (xat) [inline src/data/hicdex.js:296]: unrelated to this pr, but afaik 152 is the first objkt id

File: src/data/hicdex.js
Comment (xat) [inline src/data/hicdex.js:306]: would it make sense to re-throw the error here?

**Comment by xat** [comment]
looks very nice!

**Comment by melMass** [comment]
Ok now it's ready to be reviewed for real @xat 😇

**Comment by xat** [comment]
from the first review, there are few (small) open questions, but otherwise it looks good!

maybe some non-devs can do a bit of manual testing before we merge it.

**Comment by melMass** [comment]
@xat Mmhh so while debugging the feed thing (that's actually search), it seems src/pages/feed isn't used anymore?

**Comment by xat** [comment]
> @xat Mmhh so while debugging the feed thing (that's actually search), it seems src/pages/feed isn't used anymore?

yeah, looks like `/random`, `/latest` and `/hdao` are dead routes

**Review by xat — APPROVED** [review]
can be merged imo, if enough testing was done

**Comment by xat** [comment]
nice work :>

File: package.json
Comment (melMass) [reply package.json:9]: Yep that's the whole idea of this PR

File: package.json
Comment (melMass) [reply package.json]: I'm not sure like that I'm going to check in context!

File: package.json
Comment (melMass) [reply package.json:68]: [Craco](https://github.com/gsoft-inc/craco) is what allows all I'm doing here: extend CRA without ejecting

Comment (melMass) [reply package.json:68]: Assert for the polyfills, I'm not 100% sure it's needed but if not used I don't think it adds anything to the bundle

File: src/data/hicdex.js
Comment (melMass) [reply src/data/hicdex.js]: I'm going to do a round of console.log indeed! I've seen others while checking the console.

File: src/data/hicdex.js
Comment (melMass) [reply src/data/hicdex.js:296]: As far as I can tell this value was always set to that. I wonder what is between 152 and 196 maybe tests?

File: src/data/hicdex.js
Comment (melMass) [reply src/data/hicdex.js:306]: I think one of the refactor aspects will be to standardize that. Some return undefined, some log errors, some throw!
I agree they should all throw and excpetions should be handled where it's used

File: src/data/hicdex.js
Comment (melMass) [reply src/data/hicdex.js]: Mmm I'll just remove those you're right, it will complicate rebases on top. But I think we can remove 90% of them 😅

**Pull request merged**
